src/unstract/adapters/vectordb/pinecone/src/static/json_schema.json

```diff
@@ -19,12 +19,96 @@
       "format": "password",
       "description": "Provide the API key for your Pinecone account"
     },
-    "environment": {
+    "spec": {
       "type": "string",
-      "title": "Environment",
-      "type": "string",
-      "default": "us-west1-gcp",
-      "description": "Provide the environment for your Pinecone account. Example us-west1-gcp-free"
+      "title": "Specification",
+      "enum": [
+        "pod",
+        "serverless"
+      ],
+      "default": "serverless",
+      "description": "Configurations describing how the index should be deployed"
+    }
+  },
+  "allOf": [
+    {
+      "if": {
+        "properties": {
+          "spec": {
+            "const": "pod"
+          }
+        }
+      },
+      "then": {
+        "properties": {
+          "environment": {
+            "type": "string",
+            "title": "Environment",
+            "default": "us-west1-gcp",
+            "description": "Provide the environment for your Pinecone account. Example us-west1-gcp-free"
+          },
+          "replicas": {
+            "type": "string",
+            "title": "Number of replicas",
+            "default": "1",
+            "description": "The number of replicas to deploy for the pod index.",
+            "readOnly": true
```

Comment:
@gaya3-zipstack why are these attributes `readOnly`?

Reply:
These are read only because we internally default the params to these values while creating the index. At the same time, we cannot allow user to specify or edit these values later because they are scaling params and controlling them is only partially supported by Pinecone.

Reply:
@gaya3-zipstack will there be any impact on the behaviour if the user changes these values through Pinecone directly but our adapter creates a Pinecone client with defaults of 1?

Reply:
@chandrasekharan-zipstack  That is not understood clearly as we have not tested that. That is why we would like to keep it outside Unstract.

Reply:
Keeping it outside of Unstract is fine by me @gaya3-zipstack. However by doing that we shouldn't run into a problem where our pinecone client instance has a value of 1 for number of shards but in the pinecone server the user has configured 2 (or any value for that matter)

Reply:
@chandrasekharan-zipstack We are leaving it to the customer to do this migration outside Unstract. I have only understood that dynamic scaling is allowed for vertical scaling and adding replicas. When user creates an adapter, we will always connect to the same index with the defaults. Now whether user can scale this or not outside the unstract platform is something which we have not explored.
Like I said it is an unkown area for me.

Reply:
We would like to refrain from changing defaults w.r.t deployment. Also, we do not have the support to connect to a pre-existing index today in our system. We may have to support that if we have to properly support this scaling even outside Unstract. Will need a discussion around this.
@@: @nehabagdia @arun-venkataswamy @chandrasekharan-zipstack @hari-kuriakose